modelling/src/main/java/org/axonframework/modelling/command/inspection/AggregateModel.java

```diff
@@ -39,4 +41,24 @@
      * @return The current version of the aggregate
      */
     Long getVersion(T target);
+
+    /**
+     * Gets the aggregate class based on given {@code declaredType}.
+     *
+     * @param declaredType the declared type of aggregate represented by {@link String}
+     * @return the concrete aggregate class based on {@code declaredType}, if exists
+     */
+    default Optional<Class<?>> type(String declaredType) {
+        return Optional.empty();
+    }
+
+    /**
+     * Gets the declared type based on given {@code type}.
+     *
+     * @param type the type of the aggregate
+     * @return the declared aggregate type represented by {@link String}, if exists
+     */
+    default Optional<String> declaredType(Class<?> type) {
```

Comment:
How could this not exist? Or is this only if the aggregate explicitly declares a "type", causing defaults to be used when the Optional is empty?

Reply:
It can be `null` if this model does not know of provided `type` :)